### PR TITLE
Tags entity

### PR DESCRIPTION
**Pull request by carakas**
## Type

- Enhancement

## Pull request description

Use doctrine for the tags

### CODE REVIEW

File: src/Backend/Modules/Tags/Engine/Model.php [hunk 1]
```diff
@@ -27,62 +31,46 @@ class Model
...
-
-        // make sure $ids is an array
-        $ids = (array) $ids;
+        $tagRepository = self::getTagRepository();
+        $ids = array_filter((array) $ids, '\is_integer');
 
-        // delete tags
-        $database->delete('tags', 'id IN (' . implode(',', $ids) . ')');
-        $database->delete('modules_tags', 'tag_id IN (' . implode(',', $ids) . ')');
+        $tagRepository->remove(...$tagRepository->findByIds(...$ids));
```
[the remark this answers is not in the document]
Comment (carakas) [reply src/Backend/Modules/Tags/Engine/Model.php]: it is passing the items in an array as separate function arguments

File: src/Backend/Modules/Tags/Resources/config/repositories.yml [hunk 2]
```diff
@@ -1,10 +1,12 @@
 services:
   Backend\Modules\Tags\Domain\Tag\TagRepository:
     factory: ["@doctrine.orm.entity_manager", getRepository]
```
[the remark this answers is not in the document]
Comment (carakas) [reply src/Backend/Modules/Tags/Resources/config/repositories.yml]: @jeroendesloovere we will still need to declare it in repositories.yml but I'll take a look at this, hadn't seen this before

[the remark this answers is not in the document]
Comment (jeroendesloovere) [reply src/Backend/Modules/Tags/Resources/config/repositories.yml]: @carakas "declaring" will become obsolete if auto-wiring is enabled (in symfony 3.4, 4 and above)

File: src/Backend/Modules/Tags/Domain/Tag/TagRepository.php [hunk 3]
```diff
@@ -17,9 +21,64 @@ public function flush(): void
         $this->getEntityManager()->flush();
     }
 
-    public function remove(Tag $tag): void
+    public function remove(Tag ...$tag): void
```
[the remark this answers is not in the document]
Comment (carakas) [reply src/Backend/Modules/Tags/Domain/Tag/TagRepository.php]: fixed in 0476254fc

File: src/Backend/Modules/Tags/Domain/Tag/TagRepository.php [hunk 4]
```diff
@@ -17,9 +21,64 @@ public function flush(): void
...
+            function (Tag $tag) use ($entityManager) {
+                $entityManager->remove($tag);
+            },
+            $tag
+        );
+
         $this->getEntityManager()->flush();
     }
+
+    public function findByIds(int ...$id): array
```
[the remark this answers is not in the document]
Comment (carakas) [reply src/Backend/Modules/Tags/Domain/Tag/TagRepository.php]: fixed in 0476254fc

File: src/Backend/Modules/Tags/Resources/config/repositories.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (carakas) [reply src/Backend/Modules/Tags/Resources/config/repositories.yml]: changed in e1ede45bb

[the remark this answers is not in the document]
Comment (carakas) [reply src/Backend/Modules/Tags/Resources/config/repositories.yml]: I know what autowiring is :) but we still need this (see commit mentioned above)

**Pull request merged**